tests/enterprise-contract/contract.go

```diff
@@ -55,6 +56,50 @@ var _ = framework.EnterpriseContractSuiteDescribe("Enterprise Contract E2E tests
 		GinkgoWriter.Printf("Using verify EC task bundle: %s\n", verifyECTaskBundle)
 		imageWithDigest = "quay.io/redhat-appstudio/ec-golden-image:latest"
 	})
+	Context("ec-cli command verification", func() {
+		BeforeAll(func() {
+			generator = tekton.VerifyEnterpriseContract{
+				Bundle:              verifyECTaskBundle,
+				Image:               imageWithDigest,
+				Name:                "verify-enterprise-contract",
+				Namespace:           namespace,
+				PolicyConfiguration: "ec-policy",
+				PublicKey:           fmt.Sprintf("k8s://%s/%s", namespace, publicSecretName),
+				SSLCertDir:          "/var/run/secrets/kubernetes.io/serviceaccount",
+				Strict:              false,
+				EffectiveTime:       "now",
+			}
+			pipelineRunTimeout = int(time.Duration(5) * time.Minute)
+			baselinePolicies := ecp.EnterpriseContractPolicySpec{
+				Configuration: policyConfig,
+				Sources:       policySource,
+			}
+			Expect(kubeController.CreateOrUpdatePolicyConfiguration(namespace, baselinePolicies)).To(Succeed())
+		})
+		It("verifies ec cli has error handling", func() {
+			generator.Image = "quay.io/redhat-appstudio/ec-golden-image:latest"
```

Comment:
Just for my understanding, are we relying on `quay.io/redhat-appstudio/ec-golden-image:latest` not to be signed by the key in `cosign-public-key` Secret?

Reply:
Any image and key mismatch can have this violation ""msg: No image attestations found matching the given public key" and this is test case scenario

Reply:
Yes, but do we know this will hold true always for the key-image combination we use here?

Reply:
ec-golden-image get built on the [RH01 cluster ](https://console-openshift-console.apps.stone-prd-rh01.pg1f.p1.openshiftapps.com/), while "cosign-public-key" is from a random AWS hypershift cluster. 
So it is very unlikely, "cosign-public-key"  is the key to sign ec-golden-image.